src/main/java/com/j256/simplemagic/types/BigEndianString16Type.java

```diff
@@ -16,14 +16,21 @@ public Object extractValueFromBytes(int offset, byte[] bytes) {
 				break;
 			}
 		}
-		char[] chars = new char[len / 2];
-		for (int i = 0; i < len; i++) {
+		int charsLength = len / 2;
+		char[] chars = new char[charsLength];
+		for (int i = 0; i < charsLength; i++) {
 			chars[i] = bytesToChar(bytes[i * 2], bytes[i * 2 + 1]);
 		}
 		return chars;
 	}
 
+    @Override
+    public Object isMatch(Object testValue, Long andValue, boolean unsignedType, Object extractedValue,
+            MutableOffset mutableOffset, byte[] bytes) {
+        return findOffsetMatch((TestInfo) testValue, mutableOffset.offset, mutableOffset, (char[]) extractedValue);
+    }
```

Comment:
Thanks @j256 for the quick fixings, what about this isMatch() issue? it could be just because I am not sure what is the proper way to use lestring16/bestring16, but in my cases, I try to detect a UTF-16LE inputstream, when I use something like lestring16  \<html, I intend to match <html against every two byte in the target, but currently lestring16 extends StringType and when perform isMatch it ignores the extractedValue and use the original byte array, so it will test <html against every single byte in the target, as a result, 60, 0, 104, 0, 116, 0, 109, 0, 108, 0 (a valid <html string encoded as UTF-16LE) will fail to be detected as html, what do you think? Thanks